src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -215,6 +217,27 @@ namespace xrt::core::hip
     free(host_src);
   }
 
+
+  static void
+  hip_memcpy_host2device_async(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)  
+  {
+    if (!src)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid src pointer in hipMemCpyH2DAsync");
+
+    auto hip_mem_dst = memory_database::instance().get_hip_mem_from_addr(dst);
```

Comment:
`dst` may point to somewhere in the middle of `him_mem_dst` BO? Should we not keep track of that offset when we issue the actual copy command?

Reply:
That is a good catch. @zhangchiming , I am already handling this in other memcpy functions, Please use same logic. 
ideal way is to handle this to change the signature of get_hip_mem_from_addr function. It should return pair of hip_memory and offset.

Reply:
fixed.

Reply:
Looks like you are still not handling offset. Please check the hipMemCpy functions to understand the handling of offset,

Reply:
fixed.

Reply:
I still see the issue. I am essentially talking about handling of offset https://github.com/Xilinx/XRT/blob/master/src/runtime_src/hip/api/hip_memory.cpp#L142

Reply:
Fixed.